presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +470,25 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages(int[] outputChannels)
+    {
+        return new Iterator<Page>() {
+            int currentPosition = 0;
+            PageBuilder pageBuilder = new PageBuilder(types);
+            @Override
+            public boolean hasNext()
+            {
+                pageBuilder.reset();
+                currentPosition = buildPage(currentPosition, outputChannels, pageBuilder);
+                return !pageBuilder.isEmpty();
```

Comment:
Okay, but then we need two different names. You need "get pages by index" and I need "get pages" (fast). 
* if you name your method `getPages`, mine could become `getPagesUnordered`
* if you name your method `getPagesByIndex`, mine could be just `getPages` or `getPagesUnordered`

I'd like to hear your opinion, but also 3rd party's ( @kokosing ?)